crates/biome_parser/src/lexer.rs

```diff
@@ -312,37 +312,115 @@ impl LexContext for () {
+    all_checkpoints: VecDeque<LexerCheckpoint<Kind>>,
+    /// A buffer of only non-trivia checkpoints.
+    non_trivia_checkpoints: VecDeque<LexerCheckpoint<Kind>>,
+}
+
+impl<K: SyntaxKind> Lookahead<K> {
+    /// Creates a new instance of `BufferedLookahead` with specified capacity for both buffers.
+    fn with_capacity(capacity: usize) -> Self {
+        Self {
+            all_checkpoints: VecDeque::with_capacity(capacity),
+            non_trivia_checkpoints: VecDeque::with_capacity(capacity),
+        }
+    }
+
+    /// Adds a checkpoint to the end of both buffers, if applicable.
+    ///
+    /// Non-trivia checkpoints are cloned and added to the `non_trivia_checkpoints` buffer
+    /// only if they are not considered trivia.
+    fn push_back(&mut self, checkpoint: LexerCheckpoint<K>) {
+        if !checkpoint.current_kind.is_trivia() {
+            self.non_trivia_checkpoints.push_back(checkpoint.clone());
+        }
+
+        self.all_checkpoints.push_back(checkpoint);
+    }
+
+    /// Removes and returns the first element from the `all_checkpoints` buffer.
+    /// Also removes the corresponding element from `non_trivia_checkpoints` if it's non-trivia.
+    fn pop_front(&mut self) -> Option<LexerCheckpoint<K>> {
+        if let Some(lookahead) = self.all_checkpoints.pop_front() {
+            if !lookahead.current_kind.is_trivia() {
+                self.non_trivia_checkpoints.pop_front();
+            }
+            Some(lookahead)
+        } else {
+            None
+        }
+    }
+
+    /// Retrieves a reference to a checkpoint by index from the `all_checkpoints` buffer.
+    fn get_checkpoint(&self, index: usize) -> Option<&LexerCheckpoint<K>> {
+        self.all_checkpoints.get(index)
+    }
+
+    /// Retrieves a reference to a non-trivia checkpoint by index from the `non_trivia_checkpoints` buffer.
+    fn get_non_trivia_checkpoint(&self, index: usize) -> Option<&LexerCheckpoint<K>> {
+        self.non_trivia_checkpoints.get(index)
+    }
+
+    /// Checks if there are no checkpoints.
+    fn is_empty(&self) -> bool {
+        self.all_checkpoints.is_empty()
+    }
+
+    /// Clears all checkpoints from both buffers.
+    fn clear(&mut self) {
+        self.all_checkpoints.clear();
+        self.non_trivia_checkpoints.clear();
+    }
+
+    /// Returns the number of checkpoints in the `all_checkpoints` buffer.
+    fn all_len(&self) -> usize {
+        self.all_checkpoints.len()
+    }
+
+    /// Returns the number of non-trivia checkpoints in the `non_trivia_checkpoints` buffer.
+    fn non_trivia_len(&self) -> usize {
+        self.non_trivia_checkpoints.len()
+    }
+}
+
+impl<'l, Lex: Lexer<'l>> BufferedLexer<Lex::Kind, Lex> {
     /// Creates a new [BufferedLexer] wrapping the passed in [Lexer].
     pub fn new(lexer: Lex) -> Self {
         Self {
             inner: lexer,
             current: None,
-            lookahead: VecDeque::new(),
+            lookahead: Lookahead::with_capacity(5),
```

Comment:
Why 5?

Reply:
Oh, 
I wanted to look at the performance tests, but it seems there is no effect, 
Do you think it's better to use just `new`?

Reply:
Yeah, if there's no change in performance, maybe we should stick to `new`